x/staking/keeper/keeper.go

```diff
@@ -47,7 +47,7 @@ func NewKeeper(
 		panic(fmt.Sprintf("%s module account has not been set", types.NotBondedPoolName))
 	}
 
-	return Keeper{
```

Comment:
Fundamentally this change is required so that when [InitGenesis](https://github.com/cosmos/cosmos-sdk/blob/fe5c22ebd561a1901129b74ff43893908e762955/types/module/module.go#L314) iterates over the `ModuleManager` the `StakingKeeper` is retrieved by reference (not a copy) so that the post construction [SetHooks mutation](https://github.com/cosmos/cosmos-sdk/blob/1dc27595a692f47a06557427cf1e2b2af1fe0c1d/simapp/app.go#L268-L270) will modify the correct area of memory.

This is the reason for now passing the Keeper around everywhere by reference.

Not a huge fan of this, but it seems it will be up for refactor when we get to `x/distribution` and `x/slashing`, or possibly sooner.  Those modules can provide their `StakingHooks` for this module to order as `BaseAppOptions`.

Reply:
Same here